### PR TITLE
Improve Persistence Chapter

### PR DESCRIPTION
**Pull request by linawolf**
releases: main. 11.5

### CODE REVIEW

File: Documentation/ExtensionArchitecture/Extbase/Reference/Domain/Model.rst [hunk 1]
```diff
@@ -39,8 +39,16 @@ Or public getters:
...
 
 A public getter takes precedence over a public property. Getters have the
-advantage that you can make the properties themselves private and decide
-which ones should be mutable. It is also possible to have getters for
+advantage that you can make the properties themselves protected and decide
+which ones should be mutable.
+
+.. note::
+   Making properties :php:`private` does not work in Extbase models: The super
+   classes need to access the models properties directly. If your model must
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Extbase/Reference/Domain/Model.rst:47]: I assume, you mean the properties of one model? So either "model's properties" or "properties of the model"

File: Documentation/ExtensionArchitecture/Extbase/Reference/Domain/Model.rst [hunk 2]
```diff
@@ -68,6 +76,31 @@ All relationships can be defined unidirectional or multidimensional in the model
 On the side of the relationship that can only have one counterpart you must
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Extbase/Reference/Domain/Model.rst:76]:
```suggestion
On the side of the relationship that can only have one counterpart, you must
```

File: Documentation/ExtensionArchitecture/Extbase/Reference/Domain/Model.rst [hunk 3]
```diff
@@ -68,6 +76,31 @@ All relationships can be defined unidirectional or multidimensional in the model
...
+
+Nullable relations
+------------------
+
+There are two ways to allow null for a property in PHP:
+
+Nullable property types have been introduced with PHP 7.0 and can therefore
+be used in any modern TYPO3 version:
+
+.. code-block::php
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Extbase/Reference/Domain/Model.rst:88]:
```suggestion
.. code-block:: php
```

File: Documentation/ExtensionArchitecture/Extbase/Reference/Domain/Model.rst [hunk 4]
```diff
@@ -68,6 +76,31 @@ All relationships can be defined unidirectional or multidimensional in the model
...
+   :caption: Example for a nullable property
+
+   protected ?Person $secondAuthor = null;
+
+Union types that can also be used to allow null have been introduced with
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Extbase/Reference/Domain/Model.rst:93]:
```suggestion
Union types, that can also be used to allow :php:`null`, have been introduced with
```

File: Documentation/ExtensionArchitecture/Extbase/Reference/Domain/Model.rst [hunk 5]
```diff
@@ -68,6 +76,31 @@ All relationships can be defined unidirectional or multidimensional in the model
 On the side of the relationship that can only have one counterpart you must
 decide whether it is possible to have no relationship (allow null) or not.
 
+
+Nullable relations
+------------------
+
+There are two ways to allow null for a property in PHP:
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Extbase/Reference/Domain/Model.rst:83]:
```suggestion
There are two ways to allow :php:`null` for a property in PHP:
```

File: Documentation/ExtensionArchitecture/Extbase/Reference/Domain/Persistence.rst [hunk 6]
```diff
@@ -45,9 +44,24 @@ The according TCA definition could look like that:
...
 Use arbitrary database tables with an Extbase model
-===================================================
+---------------------------------------------------
 
 It is possible to use tables that do not convey to the naming scheme mentioned
 in the last section. In this case you have to define the connection between
 the database table and the file
 :ref:`EXT:{ext_key}/Configuration/Extbase/Persistence/Classes.php <extension-configuration-extbase-persistence>`.
+
+In the following example the table :sql:`fe_users`, provided by the system
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Extbase/Reference/Domain/Persistence.rst:54]:
```suggestion
In the following example, the table :sql:`fe_users` provided by the system
```

File: Documentation/ExtensionArchitecture/Extbase/Reference/Domain/Persistence.rst [hunk 7]
```diff
@@ -45,9 +44,24 @@ The according TCA definition could look like that:
...
+extension `frontend` is used as persistence table for the model
+:php:`Administrator`. Additionally the table :sql:`fe_groups` is used to persist
+the model :php:`FrontendUserGroup`.
+
+.. include:: /CodeSnippets/Extbase/Persistence/ExtbasePersistenceClasses.rst.txt
+
+The key :php:`recordType` makes sure that the defined model is only used if the
+:ref:`type of the record <t3tca:types>` is set to
+:php:`\FriendsOfTYPO3\BlogExample\Domain\Model\Administrator`. This way the
+class will only be used for administrators but bot plain frontend users.
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Extbase/Reference/Domain/Persistence.rst:64]:
```suggestion
class will only be used for administrators but not plain frontend users.
```

File: Documentation/CodeSnippets/Extbase/Annotation/Lazy.rst.txt [hunk 8]
```diff
@@ -5,12 +5,13 @@
...
+   use TYPO3\CMS\Extbase\Annotation\ORM\Lazy;
    
    class Post extends AbstractEntity
    {
        /**
         * @var ObjectStorage<Post>
-        * @Extbase\ORM\Lazy
+        * @Lazy
         */
        protected ObjectStorage $relatedPosts;
```
Comment (brotkrueml) [inline Documentation/CodeSnippets/Extbase/Annotation/Lazy.rst.txt:16]: The property is protected and there is no getter. Does this work?

File: Documentation/CodeSnippets/Extbase/Annotation/Multiple.rst.txt [hunk 9]
```diff
@@ -5,14 +5,16 @@
...
    {
        /**
         * @var ObjectStorage<Comment>
-        * @Extbase\ORM\Lazy
-        * @Extbase\ORM\Cascade("remove")
+        * @Lazy
+        * @Cascade("remove")
         */
-       public ObjectStorage $comments;
+       protected ObjectStorage $comments;
```
Comment (brotkrueml) [inline Documentation/CodeSnippets/Extbase/Annotation/Multiple.rst.txt:18]: The property is protected and there is no getter. Does this work?

File: Documentation/CodeSnippets/Extbase/Domain/Relationship1onNUni.rst.txt [hunk 10]
```diff
@@ -5,14 +5,16 @@
...
    {
        /**
         * @var ObjectStorage<Comment>
-        * @Extbase\ORM\Lazy
-        * @Extbase\ORM\Cascade("remove")
+        * @Lazy
+        * @Cascade("remove")
         */
-       public ObjectStorage $comments;
+       protected ObjectStorage $comments;
```
Comment (brotkrueml) [inline Documentation/CodeSnippets/Extbase/Domain/Relationship1onNUni.rst.txt:18]: The property is protected and there is no getter. Does this work?

**Review by brotkrueml — CHANGES_REQUESTED** [review]

File: Documentation/CodeSnippets/Extbase/Annotation/Multiple.rst.txt (hunk 9 above)
Comment (linawolf) [reply Documentation/CodeSnippets/Extbase/Annotation/Multiple.rst.txt]: These where just excepts but as the getters and setters don't really add much infomation to the topic I made them public

**Review by tomwarwick — APPROVED** [review]

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**